src/components/Tile/Tile.js

```diff
@@ -222,6 +234,18 @@ export class ExpandableTile extends Component {
     handleClick: () => {},
   };
 
+  componentWillReceiveProps({ expanded, tileMaxHeight, tilePadding }) {
```

Comment:
Should we batch these all in one `setState` call? e.g.

```js
const nextState = {};

if (expanded !== this.props.expanded) {
  nextState.expanded = expanded;
}

// ...

this.setState(nextState);
```

Reply:
OK was wondering, and hoped that it happens automagically. Definitely can do!

Reply:
I know that React batches setState calls in `cDM`, wasn't sure about `cWRP` though.

Reply:
Thought of seeing if batch happens with `cWRP`. It appears that it's true for all React lifecycle methods. Quickly created a test case and I saw render call only happens once after `cWRP` does multiple `.setState()` calls.

Reply:
Was basing off of:

> React ensures that any setState calls that happen during componentDidMount and componentDidUpdate are flushed before the user sees the updated UI.

[Reference](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html)

Reply:
I see - BTW this is from my Google search - Which tells batching happens for all React lifecycle methods, more precisely, whenever the `setState()` call is in React call stack:

> It appears that it's true for all React lifecycle methods